src/native/eventpipe/ep-session.c

```diff
@@ -215,6 +219,47 @@ ep_session_alloc (
 	ep_exit_error_handler ();
 }
 
+void
+ep_session_remove_dangling_session_states (EventPipeSession *session)
+{
+	ep_return_void_if_nok (session != NULL);
+
+	EP_RT_DECLARE_LOCAL_THREAD_ARRAY (threads);
+	ep_rt_thread_array_init (&threads);
+
+	ep_thread_get_threads (&threads);
+
+	ep_rt_thread_array_iterator_t threads_iterator = ep_rt_thread_array_iterator_begin (&threads);
+	while (!ep_rt_thread_array_iterator_end (&threads, &threads_iterator)) {
+		EventPipeThread *thread = ep_rt_thread_array_iterator_value (&threads_iterator);
+		if (thread) {
```

Comment:
When can this be null?

Reply:
I don't think it ever can be null, we only ever add non-null threads in `ep_thread_get_threads`. I think this just keeps getting copy pasted around, anywhere we iterate threads we check for null on each entry

Reply:
This check is really not needed (at least not anymore). ep_thread_get_threads is currently only called from ep_session_suspend_write_event (before this change added a new one), and the logic in ep_thread_get_threads already makes sure threads added to the lists are not null, so only way to get null pointers in the list is corruption or future coding errors. Maybe we could change the check to an EP_ASSERT or drop it.